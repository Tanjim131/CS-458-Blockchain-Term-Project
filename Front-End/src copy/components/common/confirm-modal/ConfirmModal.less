.confirm-modal {
    top: 0;
    .modal-content-wrapper {
        text-align: center;
    }
    .width_auto {
        width: auto;
    }
    .ant-modal-title {
        font-weight: 600;
    }
}