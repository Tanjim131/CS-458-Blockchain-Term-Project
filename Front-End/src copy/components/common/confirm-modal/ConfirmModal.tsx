import React, { FC } from "react";
import { Modal } from "antd";
import Button, { ButtonType } from "../button/Button";
import translate from "../../../../i18n/translate";

import "./ConfirmModal.less";

/**
 * Interface for Common Confirm Modal.
 */
interface IConfirmModal {
  visible: boolean;
  onCancel: () => void;
  onConfirm: () => void;
  children: {};
  centered?: boolean;
  onConfirmBtnText?: string;
  title: string;
  disableConfirmBtn?: boolean;
}

/**
 * This is used for ConfirmModal
 *
 * @usage: <ConfirmModal isModalVisible={visible} onCancel={function}/>
 */
const ConfirmModal: FC<IConfirmModal> = (props: IConfirmModal) => {
  const { visible, onCancel, onConfirm, children, centered = false, title, disableConfirmBtn= false, onConfirmBtnText= translate("confirm") } = props;

  return (
    <Modal
      title={title}
      visible={visible}
      centered={centered}
      onCancel={onCancel}
      className="confirm-modal"
      footer={[
        <Button key="1" className="width_auto" children={translate("cancel")} type={ButtonType.NORMAL} onClick={onCancel} />,
        <Button key="2" className="width_auto" children={onConfirmBtnText} type={ButtonType.PRIMARY} disabled={disableConfirmBtn} onClick={onConfirm} />,
      ]}
    >
      {children}
    </Modal>
  );
};

export default ConfirmModal;
