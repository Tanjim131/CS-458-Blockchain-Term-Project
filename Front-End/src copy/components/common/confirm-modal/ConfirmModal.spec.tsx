import React from 'react';
import { shallow } from "enzyme";
import ConfirmModal from "./ConfirmModal";
import { shallowToJson } from "enzyme-to-json";

describe('ConfirmModal Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    const props = {
        visible : true,
        onCancel :mockFunc,
        onConfirm :mockFunc,
        children :"",
        centered : false,
        onConfirmBtnText : "Confirm",
        title : "title",
        disableConfirmBtn : false
    }
    beforeAll(() => {
        wrapper = shallow(
            <ConfirmModal
                {...props}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it('Check if component exists or not and capture snapshot', () => {
        wrapper.setProps({ centered : true, disableConfirmBtn: true, onConfirmBtnText:"Confirm-text" });
        expect(wrapper).toBeTruthy();
    })
})