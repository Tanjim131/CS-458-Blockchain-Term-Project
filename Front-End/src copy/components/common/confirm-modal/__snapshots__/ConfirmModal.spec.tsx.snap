// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal Component Check if component exists or not and capture snapshot 1`] = `
<Modal
  centered={false}
  className="confirm-modal"
  confirmLoading={false}
  footer={
    Array [
      <Button
        className="width_auto"
        onClick={[MockFunction]}
        type={3}
      >
        <FormattedMessage
          id="cancel"
          values={Object {}}
        />
      </Button>,
      <Button
        className="width_auto"
        disabled={false}
        onClick={[MockFunction]}
        type={0}
      >
        Confirm
      </Button>,
    ]
  }
  maskTransitionName="fade"
  okType="primary"
  onCancel={[MockFunction]}
  title="title"
  transitionName="zoom"
  visible={true}
  width={520}
/>
`;
